py/picca/tests/delta_extraction/data_tests.py

```diff
@@ -126,26 +545,726 @@ def test_data_filter_forests(self):
+
+        # create a DesiHealpix with missing Data options
+        config = ConfigParser()
+        config.read_dict({"data": {
+            "catalogue": f"{THIS_DIR}/data/QSO_cat_fuji_dark_healpix.fits.gz",
+            "keep surveys": "all",
+            "input directory": f"{THIS_DIR}/data/",
+            "out dir": f"{THIS_DIR}/results/",
+            "use non-coadded spectra": False,
+            "num processors": 1,
+        }})
+        expected_message = (
+            "Missing argument 'wave solution' required by Data"
+        )
+        with self.assertRaises(DataError) as context_manager:
+            DesiHealpix(config["data"])
+        self.compare_error_message(context_manager, expected_message)
+
+        # create a DesiHealpix with missing DesiData options
+        config = ConfigParser()
+        config.read_dict({"data": {
+            "catalogue": f"{THIS_DIR}/data/QSO_cat_fuji_dark_healpix.fits.gz",
+            "keep surveys": "all",
+            "input directory": f"{THIS_DIR}/data/",
+            "out dir": f"{THIS_DIR}/results/",
+            "use non-coadded spectra": False,
+            "num processors": 1,
+            "wave solution": "lin",
+            "delta lambda": 0.8,
+            "lambda max": 5500.0,
+            "lambda max rest frame": 1200.0,
+            "lambda min": 3600.0,
+            "lambda min rest frame": 1040.0,
+            "analysis type": "BAO 3D",
+            "minimum number pixels in forest": 50,
+            "rejection log file": "rejection.fits",
+            "minimal snr bao3d": 0.0,
+        }})
+        expected_message = (
+            "Missing argument 'blinding' required by DesiData"
+        )
+        with self.assertRaises(DataError) as context_manager:
+            DesiHealpix(config["data"])
+        self.compare_error_message(context_manager, expected_message)
+
+    def test_desi_healpix_read_data(self):
+        """Test method read_data from DesiHealpix"""
+        # run with one processor; case: only sv data
+        config = ConfigParser()
+        config.read_dict({"data": {
+            "catalogue": f"{THIS_DIR}/data/QSO_cat_fuji_dark_healpix.fits.gz",
+            "keep surveys": "all",
+            "input directory": f"{THIS_DIR}/data/",
+            "out dir": f"{THIS_DIR}/results/",
+            "num processors": 1,
+        }})
+        for key, value in defaults_desi_healpix.items():
+            if key not in config["data"]:
+                config["data"][key] = str(value)
+
+        data = DesiHealpix(config["data"])
+
+        self.assertTrue(len(data.forests) == 63)
+
+        # run with 0 processors; case: only sv data
+        config = ConfigParser()
+        config.read_dict({"data": {
+            "catalogue": f"{THIS_DIR}/data/QSO_cat_fuji_dark_healpix.fits.gz",
+            "keep surveys": "all",
+            "input directory": f"{THIS_DIR}/data/",
+            "out dir": f"{THIS_DIR}/results/",
+            "num processors": 0,
```

Comment:
to have more control over this case, maybe just use 2 processors here (as 0 gets you half of what is available)

pro: guaranteed run in multiprocessing with known number of cpus (i.e. cannot accidentally be using 1 cpu when the system only has 2), e.g. same number will be used when test is run on a notebook vs github action, minimal possible demand in terms of number of cpus
con: figuring out the exact number to use would not be tested in this case (but that's not a huge issue)

I see that some test below is using 2 instead, I'd still consider using 0 here a potentially dangerous choice... We could give the if condition that sets up the right number of cpus a pragma to not be included in coverage if that's the reason for 0 here...

Reply:
missed adding the run with 2 processors in this particular test. Thanks for pointing it out. Added a commit to fix it.